src/test/scala/NumberFormatTest.scala

```diff
@@ -0,0 +1,119 @@
+package CSVValidation
+import Errors.NumberFormatError
+import com.ibm.icu.math
+import org.scalatest.FunSuite
+
+class NumberFormatTest extends FunSuite {
+  test("it formats the number received to the specified format (zero padding)") {
+    val numberFormatObj = NumberFormat(Some("00000.0000"))
+    assert(numberFormatObj.format(21.334) === "00021.3340")
+  }
+
+  test("it should recognize 1E6 as 1 million") {
+    val numberFormatObj = NumberFormat(None)
+    assert(numberFormatObj.parse("1E6") === 1000000)
+  }
+
+  test("it should recognize NAN (Not A Number)") {
+    val numberFormatObj = NumberFormat(None)
+    assert(numberFormatObj.parse("NaN").doubleValue().isNaN)
+  }
+
+  test("it should recognize INF") {
+    val numberFormatObj = NumberFormat(None)
+    assert(numberFormatObj.parse("INF").doubleValue().isInfinity)
```

Comment:
Wasn't there an `isPositiveInfinity` method? I suspect it'd be better to use that since we want to check it is specifically +INF.

Reply:
Done